layerfetcher/layers.go

```diff
@@ -552,27 +542,14 @@ func (l *Logic) FetchAtx(ctx context.Context, id types.ATXID) error {
 }
 
 // GetAtxs gets the data for given atx ids IDs and validates them. returns an error if at least one ATX cannot be fetched.
-func (l *Logic) GetAtxs(ctx context.Context, IDs []types.ATXID) error {
-	hashes := make([]types.Hash32, 0, len(IDs))
-	for _, atxID := range IDs {
-		hashes = append(hashes, atxID.Hash32())
+func (l *Logic) GetAtxs(ctx context.Context, ids []types.ATXID) error {
+	if len(ids) == 0 {
+		return nil
 	}
-	// TODO: atx Id is currently only the header bytes - should we change it?
-	results := l.fetcher.GetHashes(hashes, fetch.ATXDB, false)
-	for hash, resC := range results {
-		res := <-resC
-		if res.Err != nil {
-			l.log.WithContext(ctx).With().Error("cannot fetch atx",
-				log.String("hash", hash.ShortString()),
-				log.Err(res.Err))
-			return res.Err
-		}
-		if !res.IsLocal {
-			err := l.getAtxResults(ctx, res.Hash, res.Data)
-			if err != nil {
-				return err
-			}
-		}
+	l.log.WithContext(ctx).With().Debug("requesting atxs from peer", log.Int("num_atxs", len(ids)))
+	hashes := types.ATXIDsToHashes(ids)
+	if errs := l.getHashes(ctx, hashes, fetch.ATXDB, l.atxHandler.HandleAtxData); len(errs) > 0 {
```

Comment:
is it a common practice to return just the first error?

Reply:
ppl tend to invent their own solutions with returning multiple errors. some wrap multiple errors with errors.Wrap but i don't really like that, as for me that implies their are in the same chain of event and is confusing. some created their own customized error to include a list of errors. in this case, i just don't care enough to do that.

my rationale here is that when i'm debugging, i'll look at the log anyway and each of the missing hash has its corresponding log. the point here is to fail the upstream Block/Proposal handler. so i took the simplest approach to just return the first error.